Code/Source/Lidar/PipelineGraph.cpp

```diff
@@ -0,0 +1,180 @@
+
+        RGL_CHECK(rgl_graph_node_add_child(m_nodes.m_pointCloudTransform, m_nodes.m_pointsFormatPublish));
+    }
+
+    PipelineGraph::PipelineGraph(PipelineGraph&& other)
+        : m_nodes{ other.m_nodes }
+        , m_isPublishingEnabled{ other.m_isPublishingEnabled }
+        , m_isCompactEnabled{ other.m_isCompactEnabled }
+    {
+        other.m_nodes = {};
+    }
+
+    PipelineGraph::~PipelineGraph()
+    {
+        if (m_nodes.m_rayPoses)
+        {
+            RGL_CHECK(rgl_graph_destroy(m_nodes.m_rayTrace));
+
+            if (!IsCompactEnabled())
+            {
+                RGL_CHECK(rgl_graph_destroy(m_nodes.m_pointsCompact));
+            }
+
+            if (!IsPublishingEnabled())
+            {
+                if (m_nodes.m_pointCloudPublish)
+                {
+                    RGL_CHECK(rgl_graph_destroy(m_nodes.m_pointCloudPublish));
+                }
+
+                RGL_CHECK(rgl_graph_destroy(m_nodes.m_pointsFormatPublish));
+            }
+
+            if (!IsNoiseEnabled())
+            {
+                if (m_nodes.m_angularNoise)
+                {
+                    RGL_CHECK(rgl_graph_destroy(m_nodes.m_angularNoise));
+                }
+
+                if (m_nodes.m_distanceNoise)
+                {
+                    RGL_CHECK(rgl_graph_destroy(m_nodes.m_distanceNoise));
+                }
+            }
+        }
+    }
+
+    void PipelineGraph::SetIsCompactEnabled(bool value)
+    {
+        if (IsCompactEnabled() == value) // Make sure to only allow state switching calls
+        {
+            return;
+        }
+
+        rgl_node_t compactNodeParent = IsNoiseEnabled() ? m_nodes.m_distanceNoise : m_nodes.m_rayTrace;
+
+        if (value)
+        {
+            RGL_CHECK(rgl_graph_node_remove_child(compactNodeParent, m_nodes.m_pointsFormat));
+            RGL_CHECK(rgl_graph_node_add_child(compactNodeParent, m_nodes.m_pointsCompact));
+            RGL_CHECK(rgl_graph_node_add_child(m_nodes.m_pointsCompact, m_nodes.m_pointsFormat));
+        }
+        else
+        {
+            RGL_CHECK(rgl_graph_node_remove_child(compactNodeParent, m_nodes.m_pointsCompact));
+            RGL_CHECK(rgl_graph_node_remove_child(m_nodes.m_pointsCompact, m_nodes.m_pointsFormat));
+            RGL_CHECK(rgl_graph_node_add_child(compactNodeParent, m_nodes.m_pointsFormat));
+        }
+
+        m_isCompactEnabled = value;
+    }
+
+    void PipelineGraph::SetIsPublishingEnabled(bool value)
+    {
+        if (IsPublishingEnabled() == value) // Make sure to only allow state switching calls
+        {
+            return;
+        }
+
+        rgl_node_t pointsTransformNodeParent = IsNoiseEnabled() ? m_nodes.m_distanceNoise : m_nodes.m_rayTrace;
+
+        if (value)
+        {
+            RGL_CHECK(rgl_graph_node_add_child(pointsTransformNodeParent, m_nodes.m_pointCloudTransform));
+            RGL_CHECK(rgl_graph_node_add_child(m_nodes.m_pointsFormatPublish, m_nodes.m_pointCloudPublish));
```

Comment:
`IsPublisherConfigured` should be checked before this operation. The connecting operation would fail when the node is not created.